src/hotspot/share/opto/node.hpp

```diff
@@ -1712,6 +1728,25 @@ class Unique_Node_List : public Node_List {
 #endif
 };
 
+// Definition must appear after complete type definition of Unique_Node_List
+template <typename Callback, typename Bypass>
+void Node::visit_uses(Callback callback, Bypass bypass) {
```

Comment:
Move this to node.cpp, then you don't need to have the `class Unique_Node_List` you added at the top.

Reply:
I can remove the forward declaration, but I can't move the definition to `node.cpp` as it is a function template? It doesn't compile on my system at least when I move it.

Reply:
Ah. I guess you are right about that. There would be ways, but you'd have to make sure to instanciate all required templates... not worth it. Leave it here.